src/Simplify.cpp

```diff
@@ -2500,6 +2641,54 @@ class Simplify : public IRMutator {
                    equal(add_a->b, add_b->a)) {
             // min(a + b, b + c) -> min(a, c) + b
             expr = mutate(min(add_a->a, add_b->b)) + add_a->b;
+        } else if (no_overflow(op->type) &&
```

Comment:
Are these all permutations of commuting/reassociating the same rule? I'm fine with it... just curious... we should try to think about a better way to do this (in a separate PR). I know we have other instances of rules like this too.